website/src/controllers/api-v3/user.js

```diff
@@ -31,4 +39,98 @@ api.getUser = {
   },
 };
 
+let partyMembersFields = 'profile.name stats achievements items.special';
+
+/**
+ * @api {post} /user/class/cast/:spell Cast a spell on a target.
+ * @apiVersion 3.0.0
+ * @apiName UserCast
+ * @apiGroup User
+ *
+ * @apiParam {string} spellId The spell to cast.
+ * @apiParam {string="party","self","user","task"} targetType Query parameter, the target type for the spell.
+ * @apiParam {UUID} targetId Optional query parameter, the id of the target when casting a spell on a party member or a task.
+ *
+ * @apiSuccess {Object|Array} mixed Will return the modified targets. For party members only the necessary fields will be populated.
+ */
+api.castSpell = {
+  method: 'POST',
+  middlewares: [authWithHeaders(), cron],
+  url: '/user/class/cast/:spell',
+  async handler (req, res) {
+    let user = res.locals.user;
+    let targetType = req.query.targetType;
+    let targetId = req.query.targetId;
+    let spellId = req.params.spellId;
+
+    // optional because not required by all targetTypes, presence is checked later if necessary
+    req.checkQuery('targetId', res.t('targetIdUUID')).optional().isUUID();
+    req.checkQuery('targetType', res.t('invalidTargetType')).isIn(['task', 'self', 'user', 'party']);
+
+    let reqValidationErrors = req.validationErrors();
+    if (reqValidationErrors) throw reqValidationErrors;
+
+    let klass = common.content.spells.special[spellId] ? 'special' : user.stats.class;
```

Comment:
Would now be a good time to discuss separating out the proper spells from spooky sparkle-type items and greeting cards? I think we could abstract most of the shared logic into a separate function, and I see no reason to keep them under the same route.